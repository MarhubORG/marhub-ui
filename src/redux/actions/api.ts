import {
  EXPORTING_IRAP_DATA,
  EXPORT_IRAP_DATA_FAILURE,
  EXPORT_IRAP_DATA_SUCCESS,
  EXPORTING_BOTPRESS_DATA,
  EXPORT_BOTPRESS_DATA_FAILURE,
  EXPORT_BOTPRESS_DATA_SUCCESS,
  LOGOUT,
} from '../constants/actionTypes';

export interface ExportingIrapDataPayload {
  startDate: string;
  endDate: string;
  selectedTemplate: string;
  emailText: string;
  irapUuidSearchText: string;
}

export interface ExportingIrapDataAction {
  type: typeof EXPORTING_IRAP_DATA;
  payload: ExportingIrapDataPayload;
}

export function exportingIrapData(
  data: ExportingIrapDataPayload
): ExportingIrapDataAction {
  return { type: EXPORTING_IRAP_DATA, payload: data };
}

export interface ExportIrapDataSuccessAction {
  type: typeof EXPORT_IRAP_DATA_SUCCESS;
  payload: object[];
}

export function exportingIrapDataSuccess(
  data: object[]
): ExportIrapDataSuccessAction {
  return {
    type: EXPORT_IRAP_DATA_SUCCESS,
    payload: data,
  };
}

export interface ExportIrapDataFailureAction {
  type: typeof EXPORT_IRAP_DATA_FAILURE;
  payload: string;
}

export function exportingIrapDataFailure(
  message: string
): ExportIrapDataFailureAction {
  return {
    type: EXPORT_IRAP_DATA_FAILURE,
    payload: message,
  };
}

export interface ExportingBotpressDataPayload {
  startDate: string;
  endDate: string;
  selectedTemplate: string;
  emailText: string;
  irapUuidSearchText: string;
}

export interface ExportingBotpressDataAction {
  type: typeof EXPORTING_BOTPRESS_DATA;
  payload: ExportingBotpressDataPayload;
}

export function exportingBotpressData(
  data: ExportingBotpressDataPayload
): ExportingBotpressDataAction {
  return { type: EXPORTING_BOTPRESS_DATA, payload: data };
}

export interface ExportBotpressDataSuccessAction {
  type: typeof EXPORT_BOTPRESS_DATA_SUCCESS;
  payload: object[];
}

export function exportingBotpressDataSuccess(
  data: object[]
): ExportBotpressDataSuccessAction {
  return {
    type: EXPORT_BOTPRESS_DATA_SUCCESS,
    payload: data,
  };
}

export interface ExportBotpressDataFailureAction {
  type: typeof EXPORT_BOTPRESS_DATA_FAILURE;
  payload: string;
}

export function exportingBotpressDataFailure(
  message: string
): ExportBotpressDataFailureAction {
  return {
    type: EXPORT_BOTPRESS_DATA_FAILURE,
    payload: message,
  };
}

export interface LogoutAction {
  type: typeof LOGOUT;
}

export function logout(): LogoutAction {
  return { type: LOGOUT };
}

export type ApiActionTypes =
  | ExportingIrapDataAction
  | ExportIrapDataSuccessAction
  | ExportIrapDataFailureAction
  | ExportingBotpressDataAction
  | ExportBotpressDataSuccessAction
  | ExportBotpressDataFailureAction
  | LogoutAction;
