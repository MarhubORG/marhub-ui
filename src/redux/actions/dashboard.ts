import {
  FETCH_ORGANIZATIONS,
  FETCH_ORGANIZATIONS_FAILURE,
  FETCH_ORGANIZATIONS_SUCCESS,
  LOGOUT,
  UPDATE_ORGANIZATION,
  UPDATE_ORGANIZATION_FAILURE,
  UPDATE_ORGANIZATION_SUCCESS,
} from '../constants/actionTypes';

export interface FetchOrganizationsAction {
  type: typeof FETCH_ORGANIZATIONS;
}

export function fetchOrganizations(): FetchOrganizationsAction {
  return { type: FETCH_ORGANIZATIONS };
}

export interface OrganizationFields {
  name: string;
  visible_fields: string[];
}

export interface Organization {
  organisation: OrganizationFields;
  id: number;
}

export interface FetchOrganizationsSuccessAction {
  type: typeof FETCH_ORGANIZATIONS_SUCCESS;
  payload: Organization[];
}

export function fetchOrganizationsSuccess(
  organizations: Organization[]
): FetchOrganizationsSuccessAction {
  return { type: FETCH_ORGANIZATIONS_SUCCESS, payload: organizations };
}

export interface FetchOrganizationsFailureAction {
  type: typeof FETCH_ORGANIZATIONS_FAILURE;
}

export function fetchOrganizationsFailure(): FetchOrganizationsFailureAction {
  return { type: FETCH_ORGANIZATIONS_FAILURE };
}

export interface LogoutAction {
  type: typeof LOGOUT;
}

export function logout(): LogoutAction {
  return { type: LOGOUT };
}

export interface UpdateOrganizationPayload {
  id: number;
  organization: Organization;
}

export interface UpdateOrganizationAction {
  type: typeof UPDATE_ORGANIZATION;
  payload: UpdateOrganizationPayload;
}

export function updateOrganization(
  organization: UpdateOrganizationPayload
): UpdateOrganizationAction {
  return {
    type: UPDATE_ORGANIZATION,
    payload: organization,
  };
}

export interface UpdateOrganizationSuccessPayload {
  id: number;
  organization: Organization;
}

export interface UpdateOrganizationSuccessAction {
  type: typeof UPDATE_ORGANIZATION_SUCCESS;
  payload: UpdateOrganizationSuccessPayload;
}

export function updateOrganizationSuccess(
  organization: UpdateOrganizationSuccessPayload
): UpdateOrganizationSuccessAction {
  return {
    type: UPDATE_ORGANIZATION_SUCCESS,
    payload: organization,
  };
}

export interface UpdateOrganizationFailureAction {
  type: typeof UPDATE_ORGANIZATION_FAILURE;
  payload: string;
}

export function updateOrganizationFailure(
  message: string
): UpdateOrganizationFailureAction {
  return {
    type: UPDATE_ORGANIZATION_FAILURE,
    payload: message,
  };
}

export type DashboardActionTypes =
  | LogoutAction
  | FetchOrganizationsAction
  | FetchOrganizationsSuccessAction
  | FetchOrganizationsFailureAction
  | UpdateOrganizationAction
  | UpdateOrganizationSuccessAction
  | UpdateOrganizationFailureAction;
