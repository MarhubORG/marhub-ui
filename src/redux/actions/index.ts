import { SIGNUP, SIGNUP_REDIRECTING } from '../constants/actionTypes';
import { Action } from '../../types/interfaces';

export function signup(
  name: string,
  organization: string,
  email: string,
  password: string
): Action {
  return { type: SIGNUP, payload: { name, organization, email, password } };
}

export function signupRedirecting(): Action {
  return { type: SIGNUP_REDIRECTING };
}
