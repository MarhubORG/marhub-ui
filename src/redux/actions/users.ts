import {
  FETCH_USERS,
  FETCH_USERS_FAILURE,
  FETCH_USERS_SUCCESS,
  CREATE_USER,
  CREATE_USER_FAILURE,
  CREATE_USER_SUCCESS,
  EDIT_USER,
  EDIT_USER_FAILURE,
  EDIT_USER_SUCCESS,
  DELETE_USER,
  DELETE_USER_FAILURE,
  DELETE_USER_SUCCESS,
  LOGOUT,
} from '../constants/actionTypes';

export interface FetchUsersAction {
  type: typeof FETCH_USERS;
}

export interface UserProfileFields {
  name: string;
  organisation: string;
  role?: string;
}

export interface User {
  id: number;
  email: string;
  isDisabled: boolean;
  profile: UserProfileFields;
}

export function fetchUsers(): FetchUsersAction {
  return {
    type: FETCH_USERS,
  };
}

export interface FetchUsersFailureAction {
  type: typeof FETCH_USERS_FAILURE;
}

export function fetchUsersFailure(): FetchUsersFailureAction {
  return {
    type: FETCH_USERS_FAILURE,
  };
}

export interface FetchUsersSuccessAction {
  type: typeof FETCH_USERS_SUCCESS;
  payload: User[];
}

export function fetchUsersSuccess(payload: User[]): FetchUsersSuccessAction {
  return {
    type: FETCH_USERS_SUCCESS,
    payload,
  };
}

export interface CreateUserPayload {
  email: string;
  name: string;
  selectedOrganization: string;
  role: string;
  password: string;
  isDisabled: boolean;
}

export interface EditUserPayload {
  email: string;
  name: string;
  selectedOrganization: string;
  role: string;
  password: string;
  isDisabled: boolean;
  id: number;
}

export interface CreateUserAction {
  type: typeof CREATE_USER;
  payload: CreateUserPayload;
}

export function createUser(data: CreateUserPayload): CreateUserAction {
  return {
    type: CREATE_USER,
    payload: {
      email: data.email,
      name: data.name,
      selectedOrganization: data.selectedOrganization,
      role: data.role,
      password: data.password,
      isDisabled: data.isDisabled,
    },
  };
}

export interface CreateUserFailureAction {
  type: typeof CREATE_USER_FAILURE;
}

export function createUserFailure(): CreateUserFailureAction {
  return {
    type: CREATE_USER_FAILURE,
  };
}

export interface CreateUserSuccessAction {
  type: typeof CREATE_USER_SUCCESS;
  payload: User;
}

export function createUserSuccess(payload: User): CreateUserSuccessAction {
  return {
    type: CREATE_USER_SUCCESS,
    payload,
  };
}

export interface EditUserAction {
  type: typeof EDIT_USER;
  payload: EditUserPayload;
}

export function editUser(payload: EditUserPayload): EditUserAction {
  return {
    type: EDIT_USER,
    payload,
  };
}

export interface EditUserSuccessAction {
  type: typeof EDIT_USER_SUCCESS;
  payload: User;
}

export function editUserSuccess(payload: User): EditUserSuccessAction {
  return {
    type: EDIT_USER_SUCCESS,
    payload,
  };
}

export interface EditUserFailureAction {
  type: typeof EDIT_USER_FAILURE;
}

export function editUserFailure(): EditUserFailureAction {
  return {
    type: EDIT_USER_FAILURE,
  };
}

export interface LogoutAction {
  type: typeof LOGOUT;
}

export function logout(): LogoutAction {
  return {
    type: LOGOUT,
  };
}

export interface DeleteUserAction {
  type: typeof DELETE_USER;
  payload: number;
}

export function deleteUser(payload: number): DeleteUserAction {
  return {
    type: DELETE_USER,
    payload,
  };
}

export interface DeleteUserFailureAction {
  type: typeof DELETE_USER_FAILURE;
}

export function deleteUserFailure(): DeleteUserFailureAction {
  return {
    type: DELETE_USER_FAILURE,
  };
}

export interface DeleteUserSuccessAction {
  type: typeof DELETE_USER_SUCCESS;
}

export function deleteUserSuccess(): DeleteUserSuccessAction {
  return {
    type: DELETE_USER_SUCCESS,
  };
}

export type UserActionTypes =
  | FetchUsersAction
  | FetchUsersFailureAction
  | FetchUsersSuccessAction
  | CreateUserAction
  | CreateUserFailureAction
  | CreateUserSuccessAction
  | EditUserAction
  | EditUserSuccessAction
  | EditUserFailureAction
  | DeleteUserAction
  | DeleteUserFailureAction
  | DeleteUserSuccessAction
  | LogoutAction;
