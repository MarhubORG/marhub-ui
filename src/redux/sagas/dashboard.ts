import axios from 'axios';
import { put, takeLatest } from 'redux-saga/effects';
import cookie from 'react-cookies';

import {
  FETCH_ORGANIZATIONS,
  UPDATE_ORGANIZATION,
  CREATE_ORGANIZATION,
  CREATE_TEMPLATE,
  UPDATE_TEMPLATE,
  DELETE_TEMPLATE,
} from '../constants/actionTypes';
import {
  fetchOrganizationsSuccess,
  fetchOrganizationsFailure,
  FetchOrganizationsAction,
  UpdateOrganizationAction,
  updateOrganizationSuccess,
  updateOrganizationFailure,
  CreateOrganizationAction,
  createOrganizationFailure,
  createOrganizationSuccess,
  CreateTemplateAction,
  createTemplateFailure,
  createTemplateSuccess,
  UpdateTemplateAction,
  updateTemplateSuccess,
  updateTemplateFailure,
  DeleteTemplateAction,
  deleteTemplateSuccess,
  deleteTemplateFailure,
} from '../actions/dashboard';

function* fetchOrgs(action: FetchOrganizationsAction): object {
  try {
    const token = cookie.load('token');
    const axiosConfig = {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };
    const url = 'http://localhost:8080/api/v1/organisations';

    const json = yield axios.get(url, axiosConfig);
    yield put(fetchOrganizationsSuccess(json.data[0]));
  } catch (error) {
    // const errorMessage = `${error.request.status} Error: ${error.response.data.error}`;
    yield put(fetchOrganizationsFailure());
  }
}

export function* fetchOrganizationsWatcher(): object {
  yield takeLatest(FETCH_ORGANIZATIONS, fetchOrgs);
}

function* updateOrganization(action: UpdateOrganizationAction): object {
  try {
    const token = cookie.load('token');
    const url = 'http://localhost:8080/api/v1/exportation_fields';
    axios.defaults.headers.common.Authorization = `Bearer ${token}`;
    const json = yield axios.post(url, {
      id: action.payload.id,
      organization: action.payload.organization,
    });
    yield put(updateOrganizationSuccess(json.data));
  } catch (error) {
    yield put(
      updateOrganizationFailure(
        'Update organization failed. Please try again or contact your administrator.'
      )
    );
  }
}

export function* updateOrganizationWatcher(): object {
  yield takeLatest(UPDATE_ORGANIZATION, updateOrganization);
}

function* createOrganization(action: CreateOrganizationAction): object {
  try {
    const token = cookie.load('token');
    const url = 'http://localhost:8080/api/v1/organisations';
    axios.defaults.headers.common.Authorization = `Bearer ${token}`;
    const json = yield axios.post(url, {
      name: action.payload,
    });
    yield put(createOrganizationSuccess(json.data));
  } catch (error) {
    yield put(createOrganizationFailure());
  }
}

export function* createOrganizationWatcher(): object {
  yield takeLatest(CREATE_ORGANIZATION, createOrganization);
}

function* createTemplate(action: CreateTemplateAction): object {
  try {
    const token = cookie.load('token');
    const url = 'http://localhost:8080/api/v1/templates';
    axios.defaults.headers.common.Authorization = `Bearer ${token}`;
    const json = yield axios.post(url, {
      name: action.payload.name,
      fields: action.payload.fields,
    });
    yield put(createTemplateSuccess(json.data));
  } catch (error) {
    yield put(createTemplateFailure());
  }
}

export function* createTemplateWatcher(): object {
  yield takeLatest(CREATE_TEMPLATE, createTemplate);
}

function* updateTemplate(action: UpdateTemplateAction): object {
  try {
    const token = cookie.load('token');
    const url = 'http://localhost:8080/api/v1/templates';
    axios.defaults.headers.common.Authorization = `Bearer ${token}`;
    const json = yield axios.post(url, {
      name: action.payload.name,
      fields: action.payload.fields,
    });
    yield put(updateTemplateSuccess(json.data));
  } catch (error) {
    const errorMessage =
      'Update template failed. Please contact the administrator.';
    yield put(updateTemplateFailure(errorMessage));
  }
}

export function* updateTemplateWatcher(): object {
  yield takeLatest(UPDATE_TEMPLATE, updateTemplate);
}

function* deleteTemplate(action: DeleteTemplateAction): object {
  try {
    const token = cookie.load('token');
    const url = 'http://localhost:8080/api/v1/templates';
    axios.defaults.headers.common.Authorization = `Bearer ${token}`;
    const json = yield axios.delete(url, {
      data: {
        name: action.payload,
      },
    });
    yield put(deleteTemplateSuccess(json.data));
  } catch (error) {
    yield put(
      deleteTemplateFailure(
        'Delete template failed. Please contact the administrator.'
      )
    );
  }
}

export function* deleteTemplateWatcher(): object {
  yield takeLatest(DELETE_TEMPLATE, deleteTemplate);
}
