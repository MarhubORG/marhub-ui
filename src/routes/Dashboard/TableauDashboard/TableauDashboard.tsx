import React, { Component } from 'react';
import styled from 'styled-components';

const REFRESHES = 'REFRESHES';

const StyledObject = styled.object`
  height: 75vh;
  width: 100vw;
`;
export default class TableauDashboard extends Component {
  componentDidMount(): void {
    const refreshes = localStorage.getItem(REFRESHES);
    if (!refreshes) {
      window.location.reload();
      localStorage.setItem(REFRESHES, 'true');
    }
  }

  componentWillUnmount(): void {
    localStorage.setItem(REFRESHES, '');
  }

  render(): JSX.Element {
    return <div>{DashboardObject()}</div>;
  }
}

export function DashboardObject(): JSX.Element {
  return (
    <div className="tableauPlaceholder" style={{ border: '1px solid black' }}>
      <StyledObject className="tableauViz">
        <param
          name="host_url"
          value="https%3A%2F%2Feu-west-1a.online.tableau.com%2F"
        />
        <param name="embed_code_version" value="3" />{' '}
        <param
          name="site_root"
          value="&#47;t&#47;marhubinternationaldashboard"
        />
        <param name="name" value="Dashboard_draft&#47;CreateTime" />
        <param name="tabs" value="yes" />
        <param name="toolbar" value="yes" />
        <param name="showAppBanner" value="false" />
      </StyledObject>
    </div>
  );
}
