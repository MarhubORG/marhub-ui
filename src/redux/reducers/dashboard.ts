import { DashboardState } from '../../types/interfaces';
import {
  FETCH_ORGANIZATIONS,
  FETCH_ORGANIZATIONS_FAILURE,
  FETCH_ORGANIZATIONS_SUCCESS,
  UPDATE_ORGANIZATION,
  UPDATE_ORGANIZATION_FAILURE,
  UPDATE_ORGANIZATION_SUCCESS,
  LOGOUT,
  CREATE_ORGANIZATION,
  CREATE_ORGANIZATION_FAILURE,
  CREATE_ORGANIZATION_SUCCESS,
  CREATE_ORGANIZATION_REDIRECT,
  CREATE_TEMPLATE,
  CREATE_TEMPLATE_FAILURE,
  CREATE_TEMPLATE_SUCCESS,
  UPDATE_TEMPLATE,
  UPDATE_TEMPLATE_FAILURE,
  UPDATE_TEMPLATE_SUCCESS,
  DELETE_TEMPLATE,
  DELETE_TEMPLATE_FAILURE,
  DELETE_TEMPLATE_SUCCESS,
  DELETE_TEMPLATE_REDIRECT,
} from '../constants/actionTypes';
import { DashboardActionTypes, Organization } from '../actions/dashboard';

export const standardFetchOrganizationErrorMessage =
  'Could not fetch organizations. Please try again later or contact the administrator.';
export const templateFailureMessage =
  'Creating template failed. Please contact your administrator.';
export const templateSuccessMessage = 'Successfully created new template';

export const initialState = {
  loading: false,
  organizations: [],
  errorMessage: '',
  redirectToVisibleFields: '',
  templateMessage: '',
};
export const successMessage = 'Successfully updated organization permissions.';
export const createOrgFailureMessage =
  'Creating a new organization failed. Please contact your administrator.';
export const createOrgSuccessMessage =
  'Successfully updated organization permissions.';
export default function dashboardReducer(
  state = initialState,
  action: DashboardActionTypes
): DashboardState {
  switch (action.type) {
    case FETCH_ORGANIZATIONS:
      return { ...state, loading: true, errorMessage: '' };
    case FETCH_ORGANIZATIONS_SUCCESS:
      return {
        ...state,
        loading: false,
        organizations: [...action.payload],
        errorMessage: '',
      };
    case FETCH_ORGANIZATIONS_FAILURE:
      return {
        ...state,
        loading: false,
        errorMessage: standardFetchOrganizationErrorMessage,
      };
    case UPDATE_ORGANIZATION:
      return { ...state, loading: true, errorMessage: '' };
    case UPDATE_ORGANIZATION_FAILURE:
      return { ...state, loading: false, errorMessage: action.payload };
    case UPDATE_ORGANIZATION_SUCCESS:
      return {
        ...state,
        loading: false,
        errorMessage: successMessage,
        organizations: replaceWithUpdatedOrg(action.payload, state),
      };
    case LOGOUT:
      return { ...initialState };
    case CREATE_ORGANIZATION:
      return {
        ...state,
        loading: true,
        errorMessage: '',
      };
    case CREATE_ORGANIZATION_FAILURE:
      return {
        ...state,
        loading: false,
        errorMessage: createOrgFailureMessage,
      };
    case CREATE_ORGANIZATION_SUCCESS:
      return {
        ...state,
        organizations: [...state.organizations, action.payload],
        loading: false,
        errorMessage: createOrgSuccessMessage,
        redirectToVisibleFields: action.payload.organisation.name,
      };
    case CREATE_ORGANIZATION_REDIRECT:
      return {
        ...state,
        redirectToVisibleFields: '',
      };
    case CREATE_TEMPLATE_FAILURE:
      return {
        ...state,
        templateMessage: templateFailureMessage,
        loading: false,
      };
    case CREATE_TEMPLATE:
      return {
        ...state,
        loading: true,
        templateMessage: '',
      };
    case CREATE_TEMPLATE_SUCCESS:
      return {
        ...state,
        loading: false,
        templateMessage: templateSuccessMessage,
        organizations: replaceWithUpdatedOrg(action.payload, state),
      };
    case UPDATE_TEMPLATE:
      return {
        ...state,
        loading: true,
        templateMessage: '',
      };
    case UPDATE_TEMPLATE_FAILURE:
      return {
        ...state,
        loading: false,
        templateMessage: action.payload,
      };
    case UPDATE_TEMPLATE_SUCCESS:
      return {
        ...state,
        loading: false,
        templateMessage: 'Updated template successfully',
        organizations: replaceWithUpdatedOrg(action.payload, state),
      };
    case DELETE_TEMPLATE:
      return {
        ...state,
        loading: true,
        templateMessage: '',
      };
    case DELETE_TEMPLATE_FAILURE:
      return {
        ...state,
        loading: false,
        templateMessage: action.payload,
      };
    case DELETE_TEMPLATE_SUCCESS:
      return {
        ...state,
        loading: false,
        templateMessage: 'Template Deleted.',
        organizations: replaceWithUpdatedOrg(action.payload, state),
      };
    case DELETE_TEMPLATE_REDIRECT:
      return {
        ...state,
        loading: false,
        templateMessage: '',
      };
    default:
      return state;
  }
}

export function replaceWithUpdatedOrg(
  organization: Organization,
  state: DashboardState
): Organization[] {
  return state.organizations.map(el => {
    if (el.id === organization.id) return organization;
    return el;
  });
}
