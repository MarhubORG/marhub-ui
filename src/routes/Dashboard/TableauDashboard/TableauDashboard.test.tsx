import React from 'react';
import renderer from 'react-test-renderer';
import TableauDashboard from './TableauDashboard';
import { createTemplate } from '../../../redux/actions/dashboard';

it('renders correctly', () => {
  const tree = renderer.create(<TableauDashboard />).toJSON();
  expect(tree).toMatchSnapshot();
});
