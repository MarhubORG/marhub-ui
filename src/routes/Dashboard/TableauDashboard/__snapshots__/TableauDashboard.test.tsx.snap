// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  height: 75vh;
  width: 100vw;
}

<div>
  <div
    className="tableauPlaceholder"
    style={
      Object {
        "border": "1px solid black",
      }
    }
  >
    <object
      className="c0 tableauViz"
    >
      <param
        name="host_url"
        value="https%3A%2F%2Feu-west-1a.online.tableau.com%2F"
      />
      <param
        name="embed_code_version"
        value="3"
      />
       
      <param
        name="site_root"
        value="/t/marhubinternationaldashboard"
      />
      <param
        name="name"
        value="Dashboard_draft/CreateTime"
      />
      <param
        name="tabs"
        value="yes"
      />
      <param
        name="toolbar"
        value="yes"
      />
      <param
        name="showAppBanner"
        value="false"
      />
    </object>
  </div>
</div>
`;
